go/arrow/array/half_float.go

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package array
+
+import (
+	"fmt"
+	"github.com/apache/arrow/go/arrow"
+	"strings"
+)
+
+// A type which represents an immutable sequence of Float16 values.
+type HalfFloat struct {
```

Comment:
for consistency with other Go types, I think I would name it `Float16` (same for the builder.)

Reply:
the problem is we already defined Float16 as a type (see line 28 type_traits_float16.go), which handles float16 to float32 conversion. However, this is in arrow package instead of array package, the concern is there might be confusion around arrow.Float16 vs. array.Float16.